src/main/java/org/spongepowered/api/event/sound/PlaySoundEvent.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.sound;
+
+import org.spongepowered.api.block.tileentity.Jukebox;
+import org.spongepowered.api.effect.sound.record.RecordType;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.Event;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+
+public interface PlaySoundEvent extends Event, Cancellable {
+
+    /**
+     * Gets the location of the sound being played.
+     *
+     * @return the location of the sound being played
+     */
+    Location<World> getLocation();
+
+    /**
+     * Called when a call to World.playBroadcastSound is made, for global sounds
```

Comment:
You shouldn't specify how you can call an event as you do, because it may not be the only way to do it, or other implementations of the server software might have other systems that fire this event. Instead, you could suggest it as examples, so that it should be something like:

```java
/**
 * Fired when a sound is broadcasted to all {@link Player}s on the server. 
 * <p>Examples of when this might be fired include:</p>
 *
 * <ul>
 * <li>When a Wither or Dragon dies;</li>
 * <li>When requested by a plugin via {@link World#playBroadcastSound()}</li>
 * </ul>
 */
```

However, I'm not convinced that you even need the examples.

You don't need to specify what happens when an event is cancelled - cancelling an event implicity means that you are preventing the action from happening.

Reply:
Noted, i think the examples are better than not having examples, but it was mostly for my self reference when implementing the events.